main.cpp

```diff
@@ -190,285 +205,320 @@ struct Safeway
         float amountOfLastOrder = 67.59f;
         int registrationDays = 133;
         void redeemGiftCard(int codeOfGiftcard = 133278645);
-        void redeemProduct(int pointsOfProduct = 700);
+        bool redeemProduct(int pointsOfProduct = 700);
         void changeProfile(int dateOfBirthday = 13);
     };
-    //1)order grocery online
     void orderGroceryOnline(Customer customer);
-    //2)refund
     double refund(int productNumber);
-    //3)collect coupons
     void collectCoupons(Customer customer);
 };
-/*
-3) Shelter of cats
-5 properties:
-    1)number of cats
-    2)distance 
-    3)amount of mony to adopt a cat
-    4)age of a cat
-    5)activity level
-3 things it can do
-    1)adopt a cat
-    2)donate money
-    3)visit the shelter on site
- */
-struct ShelterOfCats
+
+void Safeway::orderGroceryOnline(Customer customer)
+{
+    std::cout << "Customer's profile: " << customer.accountnumber << std::endl;
+}
+
+double Safeway::refund(int productNumber)
+{
+    std::cout << "Product Number: " << productNumber << std::endl;
+    std::cout << "Product Value: 1" << std::endl;
+    return 0.1;
+}
+
+void Safeway::collectCoupons(Customer customer)
+{
+    std::cout << "Coupons " << customer.membership << std::endl;
+}
+
+void Safeway::Customer::redeemGiftCard(int codeOfGiftcard)
+{
+    std::cout << "submit " << codeOfGiftcard << std::endl;
+}
+
+bool Safeway::Customer::redeemProduct(int pointsOfProduct)
+{
+    if (pointsOfProduct > 700)
+    {
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
+
+void Safeway::Customer::changeProfile(int dateOfBirthday)
+{
+    std::cout << " Current Date Of Birthday" << dateOfBirthday << std::endl;
+}
+
+struct ShelterOfCats//3
 {
-    //1)number of cats
     int numberOfCats = 5;
-    //2)distance
     float distance = 1.2f;
-    //3)amount of money to adopt a cat
     int amountOfMoneyToAdoptACat = 80;
-    //4)age of a cat
     float ageOfACat = 0.5f;
-    //5)activity level
     int activityLevel = 3;
-    //1)adopt a cat
     void adoptACat(int numberOfCat);
-    //2)donate money
-    int donateMoney(double donation);
-    //3)visit the shelter on site
+    double donateMoney(double donation);
     void visitTheShelterOnSite(std::string address);
 };
-/*
-4)TV series
-5 properties:
-    1)number of season
-    2)number of episode
-    3)duration of one episode
-    4)number of languages supported
-    5)rating
-3 things it can do
-    1)adjust the timeline
-    2)add to watchlist
-    3)watch trailer
- */
-struct TvSeries
+
+void adoptACat(int numberOfCat)
+{
+    std::cout << " Current number of cat:" << numberOfCat << std::endl;
+}
+
+double donateMoney(double donation)
+{
+    donation = 4.90;
+    return donation;
+}
```

Comment:
You are overwriting the function parameter, erasing its value.
When you pass values into functions, you are passing information.
When you assign something to that variable representing the data like you are doing here, you lose that information that was stored in the variable.
Whatever value `donation` had before line 274 is gone once line 274 is executed.

If this is intentional, leave the code as is. 
If you were not aware of this, then change your code.
Maybe write a `std::cout` statement instead.
- [ ] think about it and revise if necessary.